ipa-core/src/cli/crypto/hybrid_encrypt.rs

```diff
@@ -71,28 +89,199 @@ impl HybridEncryptArgs {
             panic!("could not load network file")
         };
 
-        let shares: [Vec<HybridReport<BreakdownKey, TriggerValue>>; 3] =
-            input.iter::<TestHybridRecord>().share();
+        let mut worker_pool = ReportWriter::new(key_registries, &self.output_dir);
+        for (report_id, record) in input.iter::<TestHybridRecord>().enumerate() {
+            worker_pool.submit(report_id, record.share())?;
+        }
+
+        worker_pool.join()?;
+
+        let elapsed = start.elapsed();
+        tracing::info!(
+            "Encryption process is completed. {}s",
+            elapsed.as_secs_f64()
+        );
+
+        Ok(())
+    }
+}
 
-        for (index, (shares, key_registry)) in zip(shares, key_registries).enumerate() {
-            let output_filename = format!("helper{}.enc", index + 1);
-            let mut writer = OpenOptions::new()
+/// A thread-per-core pool responsible for encrypting reports in parallel.
+/// This pool is shared across all writers to reduce the number of context switches.
+struct EncryptorPool {
+    pool: Vec<(Sender<EncryptorInput>, JoinHandle<UnitResult>)>,
+    next_worker: usize,
+}
+
+impl EncryptorPool {
+    pub fn with_worker_threads(
+        thread_count: usize,
+        file_writer: [Sender<EncryptorOutput>; 3],
+        key_registries: [KeyRegistry<PublicKeyOnly>; 3],
+    ) -> Self {
+        Self {
+            pool: (0..thread_count)
+                .map(move |i| {
+                    let (tx, rx) = channel::<EncryptorInput>();
```

Comment:
`rx` is an `EncryptorInput`, but what is `tx`? a slightly more descriptive naming here would be helpful.

Reply:
it is common in Rust to name sending and receiving ends of a channel as `tx` (transmission) and rx (receiving): https://doc.rust-lang.org/std/sync/mpsc/index.html

Reply:
ahh ok. i haven't encountered this before.